ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/receive/view/ReceiveView.kt

```diff
@@ -170,73 +162,139 @@ private fun ReceiveContents(
+        derivedStateOf {
+            listOfNotNull(
+                walletAddress.unified,
+                walletAddress.sapling.takeIf { versionInfo.isTestnet },
+                walletAddress.transparent,
+            )
         }
+    }
 
-        Spacer(modifier = Modifier.height(ZcashTheme.dimens.spacingSmall))
+    val pagerState = rememberPagerState { state.size }
 
-        Address(
-            walletAddress = walletAddress.unified,
-            onAddressCopyToClipboard = onAddressCopyToClipboard,
-            onQrImageShare = onQrImageShare,
+    Column(
+        modifier = modifier,
+    ) {
+        AddressTabs(
+            modifier = Modifier.fillMaxWidth(),
+            pagerState = pagerState,
+            tabs = state,
         )
-
-        if (versionInfo.isTestnet) {
-            Spacer(modifier = Modifier.height(ZcashTheme.dimens.spacingHuge))
-
-            Address(
-                walletAddress = walletAddress.sapling,
+        HorizontalPager(
+            modifier = Modifier.fillMaxSize(),
+            state = pagerState,
+            userScrollEnabled = false
+        ) { index ->
+            AddressPage(
+                modifier = Modifier.fillMaxSize(),
+                walletAddress = state[index],
                 onAddressCopyToClipboard = onAddressCopyToClipboard,
                 onQrImageShare = onQrImageShare,
             )
         }
+    }
+}
 
-        Spacer(modifier = Modifier.height(ZcashTheme.dimens.spacingHuge))
-
-        Address(
-            walletAddress = walletAddress.transparent,
-            onAddressCopyToClipboard = onAddressCopyToClipboard,
-            onQrImageShare = onQrImageShare,
-        )
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+private fun AddressTabs(
+    modifier: Modifier,
+    pagerState: PagerState,
+    tabs: List<WalletAddress>,
+    coroutineScope: CoroutineScope = rememberCoroutineScope(),
+    onTabSelected: (index: Int) -> Unit = {},
+) {
+    TabRow(
+        modifier = modifier
+            .padding(horizontal = 64.dp)
+            .border(4.dp, Color.Black),
+        selectedTabIndex = pagerState.currentPage,
+        divider = {},
+        indicator = {},
+    ) {
+        tabs.forEachIndexed { index, tab ->
+            AddressTab(
+                walletAddress = tab,
+                selected = pagerState.currentPage == index,
+                onClick = {
+                    coroutineScope.launch {
+                        onTabSelected(index)
+                        pagerState.animateScrollToPage(index)
+                    }
+                },
+            )
+        }
     }
 }
 
-private val DEFAULT_QR_CODE_SIZE = 320.dp
+@Composable
+private fun AddressTab(
+    walletAddress: WalletAddress,
+    selected: Boolean,
+    onClick: () -> Unit,
+) {
+    Tab(
+        modifier = Modifier
+            .fillMaxWidth(),
+        selected = selected,
+        onClick = onClick,
+        selectedContentColor = Color.Black,
```

Comment:
We could rather use the predefined colors from `ZcashTheme.colors` to make it easier for us once we finish the dark mode integration.

Reply:
Totally. In general typo & colors should be taken from theme instead of being hardcoded like this